xtevent/_eventols.ado

```diff
@@ -783,5 +926,105 @@ program define parsesavek, rclass
 	return local savekl "`anything'"
 	return local noestimatel "`noestimate'"
 	return local saveintl "`saveinteract'"
-end	
+end
+
+* program to parse cohort
+program define parsecohort, rclass
+
+	syntax namelist(min=1 max=2), [force] [save] [replace]
+
+	loc words=wordcount("`namelist'")
+
+	if `words' == 1 {
+		if "`namelist'"=="create" {
+			loc cohortvar = ""
+			loc cohorttype = "create"
+		}
+		else {
+			confirm variable `namelist'		
+			di as text _n "Warning: Using old syntax for cohort. New syntax for using a cohort variable is cohort(variable varname)"
+			di as text "The old syntax will be deprecated in the next version of {cmd:xtevent}"
+			loc cohortvar "`namelist'"
+			loc cohorttype "variable"
+		}
+	}
+	else if `words'==2 {
+		loc first: word 1 of `namelist'
+		if ("`first'" != "variable")  {
+			di as err _n "Invalid syntax for cohort"
+			exit 301
+		}
+		loc second : word 2 of `namelist'
+		loc cohortvar "`second'"
+		loc cohorttype "variable"		
+	}
+
+	if "`cohorttype'"=="create" & "`force'"!="" {
+		di _n "cohort variable created, force option ignored"
+	}
+
+	if ("`save'"!="" | "`replace'"!="") & "`cohorttype'"=="variable" {
+		di as err _n "cohort variable can only be saved/replaced if created"
+		exit 301
+	}
+
+	if "`replace'"!="" & "`save'"=="" {
+		di as err _n "cohort variable replace only allowed with save"
+		exit 301
+	}
+
+	return local cohortvar = "`cohortvar'"
+	return local cohorttype = "`cohorttype'"
+	return local force = "`force'"
+	return local save = "`save'"
+	return local replace = "`replace'"
+end
+
+* program to parse control_cohort
+program define parsecontrol_cohort, rclass
+
+	syntax namelist(min=1 max=2), [save] [replace]
+
+	loc words=wordcount("`namelist'")
+
+	if `words' == 1 {
+		if "`namelist'"=="create" {
+			loc control_cohortvar = ""
+			loc control_cohorttype = "create"
+		}
+		else {
+			confirm variable `namelist'		
+			di as text _n "Warning: Using old syntax for control_cohort. New syntax for using a control cohort variable is"
+			di as text "control_cohort(variable varname)"
+			di as text "The old syntax will be deprecated in the next version of {cmd:xtevent}"
+			loc control_cohortvar "`namelist'"
+			loc control_cohorttype "variable"
+		}
+	}
+	else if `words'==2 {
+		loc first: word 1 of `namelist'
+		if ("`first'" != "variable")  {
+			di as err _n "Invalid syntax for control_cohort"
+			exit 301
+		}
+		loc second : word 2 of `namelist'

```

Comment:
@jorpppp 
If `cohort` and `control_cohort` are both given, consistency checks are carried out later, but maybe we can also add some general checks in the parser for both variables. For instance, to check that `control_cohort` is binary.

Reply:
@Constantino-Carreto-Romero Since both `cohort` and `control_cohort` may also allow `create` here, I don't think here is where we should check for variable consistency. There is a check for whether the control_cohort variable is binary later, right?

Reply:
@jorpppp 
Yes, there is a check later, I had overlooked it. https://github.com/JMSLab/xtevent/blob/2cc36655978cf62b79bba3030a329dc149af63b2/xtevent/_eventols.ado#L166